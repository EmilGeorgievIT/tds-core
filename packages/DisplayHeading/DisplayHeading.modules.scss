@import '../../src/scss/settings/colours';
@import '../../src/scss/settings/typography';
@import '../../src/scss/utility/responsive';

.sup {
  composes: baseSupSubScripts from '../../src/components/Typography/Text/Text.modules.scss';
}

.base {
  composes: noSpacing from '../../src/components/Spacing.modules.scss';
  composes: wordBreak from '../../src/components/Typography.modules.scss';

  @include helvetica-neue-thin-35;
}

.heading {
  composes: base;

  font-size: 2.75rem;
  line-height: 1.14; // 48px

  .sup {
    font-size: 1.25rem;
    top: -1.2em;
  }
}

.headingDesktop {
  composes: base;

  font-size: 4.5rem;
  line-height: 1.11; // 80px
  letter-spacing: 0.2px;

  .sup {
    font-size: 1.25rem;
    top: -2.2em;
  }
}

.inverted {
  color: $color-white;
}

.default {
  color: $color-secondary;
}
