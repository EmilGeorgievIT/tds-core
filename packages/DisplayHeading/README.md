# TDS Core: DisplayHeading

Large page titles. Renders an HTML `<h1>` element.
