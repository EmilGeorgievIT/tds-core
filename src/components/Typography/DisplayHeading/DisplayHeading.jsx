import React from 'react'
import PropTypes from 'prop-types'
import safeRest from '../../../utils/safeRest'

import DisplayHeadingSup from './DisplayHeadingSup/DisplayHeadingSup'
import DisplayHeadingSub from './DisplayHeadingSub/DisplayHeadingSub'

import styles from './DisplayHeading.modules.scss'

const getClassName = invert => (
  invert ? styles.inverted : styles.default
)

/**
 * Large page titles. Renders an HTML `<h1>` element.
 *
 * <span class="docs--badge__new">new!</span> <span class="docs--badge__version">v0.22.0</span>
 */
const DisplayHeading = ({ invert, children, ...rest }) => (
  <h1 {...safeRest(rest)} className={getClassName(invert)} >{children}</h1>
)

DisplayHeading.propTypes = {
  /**
   * Invert the text color to appear light on dark backgrounds.
   */
  invert: PropTypes.bool,
  /**
   * The text. Can be text, other components, or HTML elements.
   */
  children: PropTypes.node.isRequired
}

DisplayHeading.defaultProps = {
  invert: false
}

DisplayHeading.Sup = DisplayHeadingSup
DisplayHeading.Sub = DisplayHeadingSub

export default DisplayHeading
