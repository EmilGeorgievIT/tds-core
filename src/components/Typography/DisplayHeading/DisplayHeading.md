### Minimal usage

```
<DisplayHeading>Great Deals</DisplayHeading>
```

#### Usage criteria

- Display Heading outputs an `h1` tag.
- Use as an overlay on hero, promo or headline blocks.
