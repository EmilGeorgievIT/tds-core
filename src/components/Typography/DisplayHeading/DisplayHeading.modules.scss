@import '../../../scss/settings/colours';
@import '../../../scss/settings/typography';
@import '../../../scss/settings/responsive-grid';
@import '../../../scss/utility/responsive';

.baseHeadingSupSubScripts {
  @include telus-font-family;
  position: relative;
  vertical-align: baseline;
}

.sup {
  composes: baseHeadingSupSubScripts;
}

.sub {
  composes: baseHeadingSupSubScripts;
}

.displayHeading {
  composes: noSpacing from '../../Spacing.modules.scss';

  @include helvetica-neue-thin-35;

  font-size: 2.75rem;
  line-height: 1.14; // 48px

  .sup, .sub {
    font-size: 1.25rem;
  }

  .sup {
    top: -1.2em;
  }

  .sub {
    bottom: -.5em;
  }

  @include from-breakpoint(medium) {
    font-size: 4.5rem;
    line-height: 1.11; // 80px
    letter-spacing: .2px;

    .sup {
      top: -2.2em;
    }
  }
}

.inverted {
  composes: displayHeading;

  color: $color-white;
}

.default {
  composes: displayHeading;

  color: $color-secondary;
}
