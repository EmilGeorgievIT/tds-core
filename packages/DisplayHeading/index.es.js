import './dist/index.css'
import DisplayHeading from './dist/index.es'

export default DisplayHeading
