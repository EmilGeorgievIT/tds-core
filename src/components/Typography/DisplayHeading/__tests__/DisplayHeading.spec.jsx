import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import DisplayHeading from '../DisplayHeading'

describe('DisplayHeading', () => {
  const doShallow = ({ ...props }) => shallow(
    <DisplayHeading {...props}>Great Deals</DisplayHeading>
  )

  it('renders', () => {
    const heading = doShallow()

    expect(toJson(heading)).toMatchSnapshot()
  })

  it('renders an h1', () => {
    const displayHeading = doShallow()

    expect(displayHeading).toHaveTagName('h1')
  })

  it('can be inverted', () => {
    let displayHeading = doShallow({ invert: true })
    expect(displayHeading).toHaveClassName('inverted')

    displayHeading = doShallow()
    expect(displayHeading).toHaveClassName('default')
  })

  it('passes additional attributes to h1 element', () => {
    const displayHeading = doShallow({ id: 'the-heading', tabindex: 1 })

    expect(displayHeading).toHaveProp('id', 'the-heading')
    expect(displayHeading).toHaveProp('tabindex', 1)
  })

  it('does not allow custom CSS', () => {
    const displayHeading = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(displayHeading).not.toHaveProp('className', 'my-custom-class')
    expect(displayHeading).not.toHaveProp('style')
  })
})
