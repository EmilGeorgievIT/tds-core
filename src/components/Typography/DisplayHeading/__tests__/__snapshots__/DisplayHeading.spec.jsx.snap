// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DisplayHeading renders 1`] = `
<h1
  className="default"
>
  Great Deals
</h1>
`;
